core/trino-main/src/main/java/io/trino/execution/scheduler/BroadcastOutputBufferManager.java

```diff
@@ -19,57 +19,46 @@
 import javax.annotation.concurrent.GuardedBy;
```

Comment:
Can you describe what's the purpose of this refactoring?

Reply:
It is needed by the next commit where I remove `StageLinkage` and create `PipelinedStageExecution` that now covers all pipelines semantics in one place (instead of being split between `StageLinkage` and `SqlStageExecution`). The callback way of implementing it didn't play well with the new `PipelinedStageExecution` structure.